keps/0009-operator-toolkit.md

```diff
@@ -0,0 +1,287 @@
+
+KUDO additionally supports Kustomize for defining resources and patching for tasks. Kustomize is applied after any Go templating steps. This is useful for defining common bases for objects, or for extending frameworks as described in [Extensions and Bases](#extensions-and-bases).
+
+#### Task Application
+
+Once KUDO has assembled a full set of templates for a task, they will be applied using Kubernetes server-side apply at the same time.
+
+### Parameters
+
+Parameters are a key-value list of parameter names and their defaults. As described in [Execution State](#execution-state), they are wrapped into the `.Params` object for use by all templated objects.
+
+### Templates
+
+A template is a standard Kubernetes manifest which **MAY** have additional Go Templating. These Go Templates include [Sprig](https://github.com/masterminds/sprig) to provide a standard library of functions familiar to Helm users. Additional values are available and are described in [Execution State](#execution-state).
+
+## Extensions and Bases
+
+Extensions and bases describe a mechanism for building operators or extensions to operators from a given base. As an example, a base can be an operator, a Helm chart, a CNAB bundle, or any future format that describes the deployment of a set of resources.
+
+In this document an **extension** is any KUDO operator that extends from some base. A **base** is the complete set of manifests, metadata, and other files provided by that base's type. A base should provide the complete information that users of that base tool are expected to have. The base types and what they expose to charts that extend from them are described in their respective sub-sections.
+
+To support extending from a base, `operator.yaml` is extended to support the `extends` keyword:
+
+```yaml
+extends:
+  kudo:
+    operator: "mysql"
+    version: "5.7"
+```
+
+After extending, the base resources are inherited by the extending operator. The behavior of extensions and values available from the base are described in their corresponding sub-section.
+
+When a task is defined in an extended plan, it **replaces** the task from the base. The plans available are dependent on the base type and are described more in detail in their corresponding sub-section. To support extensibility of tasks that have been replaced, a few new keywords are available in `operator.yaml` to support further control over extensions:
+
+### Tasks
+
+`task.from`: The `from` directive inside of a named task copies over all resources and patches from the base task. Resources and patches that overlap with base resource and patch names replace resource and patches already defined by the base.
+`base/`: The `base/` directive in a template reference resolves to the named template within the extended base. For example, `base/deployment.yaml` corresponds to `deployment.yaml` in the base. This enables base templates to be used directly in new plans defined by the extending operator.
```

Comment:
> `deployment.yaml` in the base
in the base framework.  

Let's be overly specific since we're using base in both places

Reply:
"For example, `base/deployment.yaml` corresponds to `deployment.yaml` file located within the base referenced by `extends`"?